### PR TITLE
SF-1315: Allow users to set query property within the store to empty string through requests

### PR DESCRIPTION
**Pull request by JoeyDeol**

### CODE REVIEW

File: packages/@storefront/core/test/unit/services/urlUtils.ts [hunk 1]
```diff
@@ -689,6 +689,60 @@ suite('URL Service', ({ expect, spy, stub }) => {
...
+
+        const newQueryState = Utils.mergeSearchQueryState(state, request);
+
+        expect(newQueryState).to.eql({
+          a: 'b',
+          original: 'foo'
+        });
+      });
+
+      it('should merge search query state properly when given a new request with a query property of empty string', () => {
```
[the remark this answers is not in the document]
Comment (carloferrer) [reply packages/@storefront/core/test/unit/services/urlUtils.ts]: 😮

File: packages/@storefront/core/test/unit/services/urlUtils.ts [hunk 2]
```diff
@@ -689,6 +689,60 @@ suite('URL Service', ({ expect, spy, stub }) => {
       });
     });
 
+    describe.only('mergeSearchQuery', () => {
```
Comment (jmbtutor) [inline packages/@storefront/core/test/unit/services/urlUtils.ts]: Take off this `.only`.

File: packages/@storefront/core/src/services/urlUtils.ts [hunk 3]
```diff
@@ -245,7 +245,7 @@ namespace UrlUtils {
 
   export const mergeSearchQueryState = (state: Store.State, request: UrlBeautifier.SearchUrlState) => ({
     ...state.data.present.query,
-    original: request.query || Selectors.query(state),
+    original: request.query === '' ? '' : request.query || Selectors.query(state),
```
Comment (jmbtutor) [inline packages/@storefront/core/src/services/urlUtils.ts]: I would check for `== undefined` instead. Better to check for the fallback case than to check for a special case.

File: packages/@storefront/core/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] [patch]
+### Changed
+- SF-1315: Allow users to set `query` to empty string within the application store.
```
Comment (jmbtutor) [inline packages/@storefront/core/CHANGELOG.md]: You should mention something about this being in the URL service, specifically the parser.

File: packages/@storefront/core/test/unit/services/urlUtils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoeyDeol) [reply packages/@storefront/core/test/unit/services/urlUtils.ts]: Lint appeased.

File: packages/@storefront/core/CHANGELOG.md (hunk 4 above)
Comment (JoeyDeol) [reply packages/@storefront/core/CHANGELOG.md]: Will do.

**Comment by JoeyDeol** [comment]
Demo Passed.

**Comment by JoeyDeol** [comment]
Leaving a Do Not Merge label here until we can pull this change into the BJs implementation in order to solve a bug revealed by the new angular changes. 

Note: a potential change will have to be made within the BJs project as well in order to ensure this release is used properly.

**Comment by JoeyDeol** [comment]
Decided to release this anyways as it would be a bug for more users beyond BJs.

**Pull request merged**